### PR TITLE
stm32wb: Add BLE support through IPM HCI driver

### PR DESCRIPTION
**Pull request by erwango**
Add BLE support on nucleo_wb55rg thanks to indroduction of HCI driver.

This pull request is an early work to get feedback on HCI driver implementation.

Based on #13742

EDIT: Added wip flash driver to avoid warnings when using `sample/bluetooth/peripheral`

### CODE REVIEW

File: drivers/bluetooth/hci/ipm_stm32wb.c [hunk 1]
```diff
@@ -0,0 +1,403 @@
...
+static K_SEM_DEFINE(ipm_busy, 1, 1);
+
+struct aci_set_tx_power {
+	u8_t cmd;
+	u8_t value[2];
+};
+
+#define ACI_WRITE_SET_TX_POWER_LEVEL       BT_OP(BT_OGF_VS, 0xFC0F)
+
+void stm32wb_start_ble(void)
```
Comment (jhedberg) [inline drivers/bluetooth/hci/ipm_stm32wb.c]: Seems like this should be static?

**Review by jhedberg — CHANGES_REQUESTED** [review]
Some initial feedback on the HCI driver

File: drivers/bluetooth/hci/ipm_stm32wb.c [hunk 2]
```diff
@@ -0,0 +1,403 @@
...
+	buf = bt_hci_cmd_create(ACI_WRITE_SET_TX_POWER_LEVEL, 3);
+	if (!buf) {
+		return -ENOBUFS;
+	}
+	param = net_buf_add(buf, sizeof(*param));
+	param->cmd = 0x0F;
+	param->value[0] = 0x01;
+	param->value[1] = 0x18;
+
+	err = bt_hci_cmd_send_sync(ACI_WRITE_SET_TX_POWER_LEVEL, NULL, &rsp);
```
[the remark this answers is not in the document]
Comment (erwango) [reply drivers/bluetooth/hci/ipm_stm32wb.c]: Right, fixed, txs

**Comment by erwango** [comment]
@jhedberg , txs for initial review, comments fixed.

**Pull request closed without merging**